lnwallet/channel.go

```diff
@@ -2993,6 +3008,50 @@ func (lc *LightningChannel) evaluateHTLCView(view *HtlcView,
 			)
 			if rmvHeight == 0 {
 				switch {
+				// If this a noop add, then when we settle the
+				// HTLC, we actually credit the sender with the
+				// amount again, thus making it a noop. Noop
+				// HTLCs are only triggered by external software
+				// using the AuxComponents and only for channels
+				// that use the custom tapscript root.
+				case entry.EntryType == Settle &&
```

Comment:
I think the whole code of this `case` block should be a standalone function. Then we can properly unit test all the boundary conditions (e.g. positive and negative deltas, balances very close to the reserve, very large or very small deltas and so on).